wsd/FileServer.cpp

```diff
@@ -494,24 +490,20 @@ void FileServerRequestHandler::sendError(int errorCode, const Poco::Net::HTTPReq
                                          const std::string& shortMessage, const std::string& longMessage,
                                          const std::string& extraHeader)
 {
-    Poco::URI requestUri(request.getURI());
-    const std::string& path = requestUri.getPath();
-    std::ostringstream oss;
-    oss << "HTTP/1.1 " << errorCode << "\r\n"
-        "Content-Type: text/html charset=UTF-8\r\n"
-        "Date: " << Util::getHttpTimeNow() << "\r\n"
-        "User-Agent: " << WOPI_AGENT_STRING << "\r\n"
-        << extraHeader
-        << "\r\n";
+    std::string body = "";
+    std::string headers = extraHeader;
     if (!shortMessage.empty())
     {
+        Poco::URI requestUri(request.getURI());
+        const std::string& path = requestUri.getPath();
         std::string pathSanitized;
         Poco::URI::encode(path, "", pathSanitized);
-        oss << "<h1>Error: " << shortMessage << "</h1>"
-            "<p>" << longMessage << ' ' << pathSanitized << "</p>"
+        headers += "Content-Type: text/html charset=UTF-8\r\n";
+        body = "<h1>Error: " + shortMessage + "</h1>" +
```

Comment:
Why can't we declare body here instead of the top of the function? Placing variables closest to the site of usage is a good practice. Also, it can be made const here.

Reply:
I think then we would have to duplicate the final sendError call, right? I don't think that would be preferable.